src/boot/efi/cpio.c

```diff
@@ -335,8 +334,14 @@ EFI_STATUS pack_cpio(
         if (!root)
                 return log_error_status_stall(EFI_LOAD_ERROR, L"Unable to open root directory.");
 
-        extra_dir_path = xpool_print(L"%D" EXTRA_DIR_SUFFIX, loaded_image->FilePath);
-        err = open_directory(root, extra_dir_path, &extra_dir);
+        if (!dropin_dir) {
+                rel_dropin_dir = xpool_print(L"%D.extra.d", loaded_image->FilePath);
```

Comment:
We've since switched to non-failing allocators (hence the x), so the OOM check can be dropped. Also maybe make both assignments in one line.

Reply:
Done